ludwig/data/dataset/ray.py

```diff
@@ -375,7 +375,7 @@ def _fetch_next_batch(self):
             self._next_batch = next(self.dataset_batch_iter)
             # If the batch has only one row and self.ignore_last, skip the batch
             # to prevent batchnorm / dropout related Torch errors
-            if self.ignore_last and len(self._next_batch[self._sample_feature_name]) == 1:
+            if self.batch_size > 1 and self.ignore_last and len(self._next_batch[self._sample_feature_name]) == 1:
```

Comment:
nice catch! i noticed this with my prompt tuning PR as well that we go into an infinite loop with batch size 1

Reply:
Yeah, this was casuing me problems today!